src/store/middlewares/roomMiddleware.tsx

```diff
@@ -53,8 +54,14 @@ const createRoomMiddleware = ({
 									);
 									dispatch(webrtcActions.setIceServers(turnServers));
 									dispatch(webrtcActions.setRTCStatsOptions(rtcStatsOptions));
+									// dispatch(webrtcActions.setClientMonitorConfig(clientMonitorConfig));
 									dispatch(roomActions.setState('joined'));
 									dispatch(joinRoom());
+
+									const roomId = getState().room.name;
+									
+									mediaService.getMonitor()?.setRoomId(roomId);
```

Comment:
Does this need to have a unique room session ID, or just the name of the room that will be the same across several sessions?

Reply:
room is enough, as observer can distinguish different sessions automatically.
If we want that the sessionId to be the same on the observed calls (sessions are called calls in observertc), then we can set it by calling `mediaService.getMonitor()?.setCallId(sessionId);`. In this case observer will need to be deployed with a config option `callIdAssignMode: SLAVE`, so observer knows the calls are dictated by the samples it receives.

Reply:
You are aware of the difference and have made an informed choice, good enough for me :smile:

Reply:
I think providing sessionId and make it consistent for the signalling server and to be later on trackable from the observed stats is important, so I need your help how can I retrieve here the sessionId